multi-node-test.sh

```diff
@@ -148,114 +107,106 @@ df -h" 2>&1 | dshbak -c; then
-}
+
+# apply fix for https://github.com/avocado-framework/avocado/issues/2908
+sudo ed <<EOF /usr/lib/python2.7/site-packages/avocado/core/runner.py
+/TIMEOUT_TEST_INTERRUPTED/s/[0-9]*$/60/
+wq
+EOF
+# apply fix for https://github.com/avocado-framework/avocado/pull/2922
+if grep \"testsuite.setAttribute('name', 'avocado')\" \
+    /usr/lib/python2.7/site-packages/avocado/plugins/xunit.py; then
+    sudo ed <<EOF /usr/lib/python2.7/site-packages/avocado/plugins/xunit.py
+/testsuite.setAttribute('name', 'avocado')/s/'avocado'/os.path.basename(os.path.dirname(result.logfile))/
+wq
 EOF
-    rm -rf $log_base_path/
-    python3 test_runner config=scripts/cart_multi_five_node.cfg \\
-        ${JENKINS_TEST_LIST_5[*]} || {
-        rc=\${PIPESTATUS[0]}
-        echo \"Test exited with \$rc\"
-    }
-    find $log_base_path/testRun -name subtest_results.yml \\
-         -exec grep -Hi fail {} \\;
 fi
+
+# put yaml files back
+restore_dist_files() {
+    local dist_files=\"\$*\"
+
+    for file in \$dist_files; do
+        if [ -f \"\$file\".dist ]; then
+            mv -f \"\$file\".dist \"\$file\"
+        fi
+    done
+
+}
+
+# set our machine names
+mapfile -t yaml_files < <(find \"$TESTDIR\" -name \\*.yaml)
+
+trap 'set +e; restore_dist_files \"\${yaml_files[@]}\"' EXIT
+
+# shellcheck disable=SC2086
+sed -i.dist -e \"s/- boro-A/- ${nodes[0]}/g\" \
+            -e \"s/- boro-B/- ${nodes[1]}/g\" \
+            -e \"s/- boro-C/- ${nodes[2]}/g\" \
+            -e \"s/- boro-D/- ${nodes[3]}/g\" \
+            -e \"s/- boro-E/- ${nodes[4]}/g\" \
+            -e \"s/- boro-F/- ${nodes[5]}/g\" \
+            -e \"s/- boro-G/- ${nodes[6]}/g\" \
+            -e \"s/- boro-H/- ${nodes[7]}/g\" \"\${yaml_files[@]}\"
+
+# let's output to a dir in the tree
+rm -rf \"$TESTDIR/avocado\" \"./*_results.xml\"
+mkdir -p "$LOGDIR"
+
+# remove test_runner dir until scons_local is updated
+rm -rf \"$TESTDIR/test_runner\"
+
+# shellcheck disable=SC2154
+trap 'set +e restore_dist_files \"\${yaml_files[@]}\"' EXIT
+
+pushd $TESTDIR
+
+# now run it!
+export PYTHONPATH=./util
+if ! ./launch.py -s \"$TEST_TAG\"; then
+    rc=\${PIPESTATUS[0]}
+else
+    rc=0
+fi
+
+mv testLogs testLogs-${1}_node
+
 exit \$rc"; then
     rc=${PIPESTATUS[0]}
 else
     rc=0
 fi
 
-scp -i ci_key -r jenkins@"${nodes[0]}:\
-$CART_BASE/install/Linux/TESTING/$log_base_path" install/Linux/TESTING/
-{
-    cat <<EOF
-TestGroup:
-    submission: $(TZ=UTC date)
-    test_group: CART_${1}-node
-    testhost: $HOSTNAME
-    user_name: jenkins
-Tests:
-EOF
-    find install/Linux/TESTING/"$log_base_path" \
-         -name subtest_results.yml -print0 | xargs -0 cat
-} > results_1.yml
-
-if ! PYTHONPATH=scony_python-junit/ \
-       jenkins/autotest_utils/results_to_junit.py; then
-    echo "Failed to convert YML to Junit"
-    if [ "$rc" = "0" ]; then
-        rc=1
-    fi
-fi
+mkdir -p install/Linux/TESTING/
+scp -i ci_key -r jenkins@"${nodes[0]}":$CART_BASE/install/Linux/TESTING/testLogs-${1}_node \
```

Comment:
(style)  line over 80 characters
(lint) Double quote to prevent globbing and word splitting. [SC2086]
(lint) Double quote to prevent globbing and word splitting. [SC2086]